cli/QuickstepCli.cpp

```diff
@@ -135,6 +140,16 @@ int main(int argc, char* argv[]) {
   gflags::ParseCommandLineFlags(&argc, &argv, true);
   google::InitGoogleLogging(argv[0]);
 
+  // Detect the hardware concurrency level.
+  unsigned int num_hw_threads = std::thread::hardware_concurrency();
+
+  // It is possible that the call above fails and returns 0, in which case
+  // the default should be 1.
+  quickstep::FLAGS_num_workers = (num_hw_threads == 0) ? 1 : num_hw_threads;
```

Comment:
This *always* overrides the user's choice for the number of threads, even if `std::thread::hardware_concurrency()` returns 0.

I think it makes sense to have a special (but still user-specifiable) value indicate that we should attempt to auto-detect the number of threads. That should also be the default for the command-line option. But, if the user explicitly specifies something else, that should take precedence over auto detection. So to summarize:

1. We change the default for `num_workers` from 1 to 0, and change the message to say that 0 (the default) means that we will attempt to auto-detect the number of threads.
2. If `FLAGS_num_workers` is nonzero, we use it as-is and do NOT attempt auto-detection.
3. If `FLAGS_num_workers` is zero, we call `std::thread::hardware_concurrency()` to attempt auto detection.
4. If `std::thread::hardware_concurrency()` returns 0, fallback on 1 thread as the default of last resort.

Or in code terms:
```
const unsigned real_num_workers
    = quickstep::FLAGS_num_workers != 0
        ? quickstep::FLAGS_num_workers
        : (std::thread::hardware_concurrency() != 0
            ? std::thread::hardware_concurrency()
            : 1);